src/screens/Sidebar/GeneralSettings.js

```diff
@@ -62,24 +110,47 @@ class GeneralSettings extends Component {
                 </Text>
                 <TextInput
                   style={{ marginTop: 4, height: 30, fontFamily: 'OpenSans', fontSize: 14 }}
-                  value={'josh@kitsu.io'}
+                  value={this.state.password}
+                  onChangeText={(t) => {
+                    this.toggle(t);
+                    this.setState({ password: t });
+                  }}
                   secureTextEntry
+                  placeholder={'Start typing to set a new password.'}
                   autoCorrect={false}
                   underlineColorAndroid={'transparent'}
                 />
               </View>
+              {this.state.shouldShowValidationInput &&
+                <View
+                  style={{ backgroundColor: 'white', paddingHorizontal: 12, paddingVertical: 8 }}
+                >
+                  <Text style={{ fontSize: 10, color: 'grey', fontFamily: 'OpenSans' }}>
+                    Current Password
+                  </Text>
+                  <TextInput
+                    style={{ marginTop: 4, height: 30, fontFamily: 'OpenSans', fontSize: 14 }}
+                    value={this.state.currentPassword}
+                    onChangeText={t => this.setState({ currentPassword: t })}
+                    secureTextEntry
+                    placeholder={'Current Password'}
+                    autoCorrect={false}
+                    underlineColorAndroid={'transparent'}
+                  />
+                </View>}
               <SidebarTitle style={{ marginTop: 20 }} title={'Content on Kitsu'} />
               <SidebarDropdown
                 title={'R18+ titles in feed, libraries, or search?'}
-                value={this.state.adult}
-                options={[{ title: 'Show Adult Titles (O_O)' }, { title: 'Hide Adult Titles' }]}
-                onSelectOption={adult => this.setState({ adult })}
+                value={this.sfwOptions[Number(this.state.sfwFilter)].title}
+                options={this.sfwOptions}
+                onSelectOption={option =>
+                  this.setState({ sfwFilter: option.title === this.sfwOptions[1].title })}
               />
               <View style={{ marginTop: 20, padding: 10, paddingLeft: 25, paddingRight: 25 }}>
```

Comment:
@edencakir would be great if you can move all the styles in this component to `StyleSheet`, that would be performant.